### PR TITLE
[HDInsight] Add parameter `--idbroker` to support customer to create ESP cluster with HDInsight Id Broker

### PR DESCRIPTION
**Pull request by aim-for-better**
**Description**  

Add parameter `--idbroker` to support customer to create ESP cluster with HDInsight Id Broker

**Testing Guide**  

**History Notes**  

[HDInsight] az hdinsight create: add deprecate information for argument `--public-networrk-access-type` and `--outbound-public-network-access-type`
[HDInsight] az hdinsight create:  add parameter `--idbroker` to support customer to create ESP cluster with HDInsight Id Broker

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
HDInsight

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py [hunk 1]
```diff
@@ -73,6 +73,9 @@ def load_arguments(self, _):
         c.argument('esp', arg_group='Cluster', action='store_true',
                    help='Specify to create cluster with Enterprise Security Package. If omitted, '
                         'creating cluster with Enterprise Security Package will not not allowed.')
+        c.argument('idbroker', arg_group='Cluster', action='store_true',
+                   help='Specify to create ESP cluster with HDInsight ID Broker. If omitted, '
+                        'creating ESP cluster with HDInsight ID broker will not not allowed.')
```
Comment (deshriva) [inline src/azure-cli/azure/cli/command_modules/hdinsight/_params.py:78]: >b [](start = 64, length = 1)

nit: B capital

**Comment by aim-for-better** [comment]
For this feature I can't add test because this feature requires some specific complicated condition.  I have tested offline many times.

**Review by Juliehzl — APPROVED** [review]

**Comment by Juliehzl** [comment]
> For this feature I can't add test because this feature requires some specific complicated condition. I have tested offline many times.

Prefer adding tests if possible. Although I trust you have tested in your machine, I will have concern on incoming SDK upgrade, etc.

**Comment by aim-for-better** [comment]
> > For this feature I can't add test because this feature requires some specific complicated condition. I have tested offline many times.
> 
> Prefer adding tests if possible. Although I trust you have tested in your machine, I will have concern on incoming SDK upgrade, etc.

It is hard for me to add test. because this is related with ESP it requires some specific identity that I can't hard code and I can't create dynamically.

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py (hunk 1 above)
Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: > > b [](start = 64, length = 1)
> 
> nit: B capital

Thanks for your comment. Fixed

**Pull request merged**